### PR TITLE
[EVM] Add bridging interface to EVM contract - stable cadence port

### PR DESCRIPTION
**Pull request by janezpodhostnik**
original PR: https://github.com/onflow/flow-go/pull/5677

### CODE REVIEW

**Comment by sisyphusSmiling** [comment]
Looks like I'm still getting errors in `bootstrap_test.go` on `expectedStateCommitment`. @janezpodhostnik @ramtinms Would the updates to the EVM contract affect the genesis state commitment or do you think this issue is unrelated to the changes here?

**Comment by janezpodhostnik** [comment]
Yes, the changes are expected due to the hash of the contract changing. 

The test have to be adjusted to use the new state commitment.

**Comment by sisyphusSmiling** [comment]
Ah makes sense. How do I generate the new hash?

**Comment by sisyphusSmiling** [comment]
@turbolent I'm not certain as I'm not familiar with the processes to manage flow-go. What would you suggest here? I don't want this PR to create conflicts down the road.

**Review by sideninja — APPROVED** [review]

**Comment by sideninja** [comment]
> Is this separate PR needed? Once the feature landed on master (#5677 got merged), the usual sync of master into the Cadence 1.0 feature branch will bring it along anyways.
> 
> Happy to already merge it, more a note that in the future, we could save effort here.

I agree, in the future is more clear to merge master.

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
#5767 is currently bringing this to `feature/stable-cadence`. Once it landed, we can update this branch and there shouldn't be really anything left

**Comment by janezpodhostnik** [comment]
I update the PR and there are some things that are missing on stable cadence branch. @sisyphusSmiling can you check if we want to merge this (I think we do)

**Review by sisyphusSmiling — APPROVED** [review]

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -348,26 +349,22 @@ contract EVM {
...
             vault: @{FungibleToken.Vault},
-            feeProvider: &{FungibleToken.Provider}
+            feeProvider: auth(FungibleToken.Withdraw) &{FungibleToken.Provider}
         ) {
-            EVM.borrowBridgeAccessor().depositTokens(
-                vault: <-vault,
-                to: self.address(),
-                feeProvider: feeProvider
-            )
+            EVM.borrowBridgeAccessor().depositTokens(vault: <-vault, to: self.address(), feeProvider: feeProvider)
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:354]:
```suggestion
            EVM.borrowBridgeAccessor().depositTokens(
                vault: <-vault,
                to: self.address(),
                feeProvider: feeProvider
            )
```

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**